packages/nimble-components/src/table/specs/README.md

```diff
@@ -133,13 +133,25 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not specified, a default ID will be generated. If the attribute is invalid, no rows in the table will be rendered, and the table will enter an invalid state according to the `validity` property and `checkValidity()` function. The attribute is invalid in the following conditions:
```

Comment:
I think not rendering for an invalid state makes sense. After sleeping on it I don't like the idea of "saving the last render", that means you can end up in states where two tables have the exact same property values assigned and show different results. Seems sketchy. Elements should be as declarative as possible.

If we revise the not rendering decision in the future I'd advocate for partial / incomplete / fallback rendering of some sort instead of saving the last render. If the id field is invalid, set the validity error on the table but go ahead and render using index mode instead. That direction opened up cans of worms but I'd rather go the "do something with the properties you have" direction than the "show the last render" direction.

Just a brain dump / thought, I dunno if there is anything to capture. Maybe we can have an open question about the behavior for invalid states and capture some note there.

Reply:
I've added some notes in the `Open Issues` section.